src/ring_theory/graded_algebra/radical.lean

```diff
@@ -0,0 +1,201 @@
+      have not_mem : i ∉ set₁ := λ h, lt_irrefl _
+        ((max'_lt_iff set₁ (nonempty x rid₁)).mp max_lt i h),
+      rw set₁_eq at not_mem,
+      simp only [not_and, not_not, ne.def, dfinsupp.mem_support_to_fun, mem_filter] at not_mem,
+      exact ideal.mul_mem_right _ I (not_mem H₂), },
+    { -- in this case  `max₂ < j`, then `yⱼ ∈ I`; for otherwise `j ∈ set₂`, then `j ≤ max₂`.
+      have not_mem : j ∉ set₂ := λ h, lt_irrefl _
+        ((max'_lt_iff set₂ (nonempty y rid₂)).mp max_lt j h),
+      rw set₂_eq at not_mem,
+      simp only [not_and, not_not, ne.def, dfinsupp.mem_support_to_fun, mem_filter] at not_mem,
+      exact ideal.mul_mem_left I _ (not_mem H₃), }, },
+
+  have not_mem_I : proj 𝒜 max₁ x * proj 𝒜 max₂ y ∉ I,
+  { have neither_mem : proj 𝒜 max₁ x ∉ I ∧ proj 𝒜 max₂ y ∉ I,
+    { rw mem_filter at mem_max₁ mem_max₂,
+      exact ⟨mem_max₁.2, mem_max₂.2⟩, },
+    intro rid,
+    cases homogeneous_mem_or_mem ⟨max₁, submodule.coe_mem _⟩ ⟨max₂, submodule.coe_mem _⟩ mem_I,
+    { apply neither_mem.1 h },
+    { apply neither_mem.2 h }, },
+
+  exact not_mem_I mem_I,
+end⟩
+
+lemma ideal.is_homogeneous.is_prime_iff {I : ideal A} (h : I.is_homogeneous 𝒜) :
+  I.is_prime ↔
+  (I ≠ ⊤) ∧
+    ∀ {x y : A}, set_like.is_homogeneous 𝒜 x → set_like.is_homogeneous 𝒜 y
+      → (x * y ∈ I → x ∈ I ∨ y ∈ I) :=
+⟨λ HI,
+  ⟨ne_of_apply_ne _ HI.ne_top, λ x y hx hy hxy, ideal.is_prime.mem_or_mem HI hxy⟩,
+  λ ⟨I_ne_top, homogeneous_mem_or_mem⟩,
+    h.is_prime_of_homogeneous_mem_or_mem I_ne_top @homogeneous_mem_or_mem⟩
+
+lemma ideal.is_prime.homogeneous_core {I : ideal A} (h : I.is_prime) :
+  (I.homogeneous_core 𝒜 : ideal A).is_prime :=
+begin
+  apply (ideal.homogeneous_core 𝒜 I).prop.is_prime_of_homogeneous_mem_or_mem,
+  { exact ne_top_of_le_ne_top h.ne_top (ideal.coe_homogeneous_core_le 𝒜 I) },
+  rintros x y hx hy hxy,
+  have H := h.mem_or_mem (ideal.coe_homogeneous_core_le 𝒜 I hxy),
+  refine H.imp _ _,
+  { exact ideal.mem_homogeneous_core_of_is_homogeneous_of_mem hx, },
+  { exact ideal.mem_homogeneous_core_of_is_homogeneous_of_mem hy, },
+end
+
+lemma ideal.is_homogeneous.radical_eq {I : ideal A} (hI : I.is_homogeneous 𝒜) :
+  I.radical = Inf { J | I ≤ J ∧ J.is_homogeneous 𝒜 ∧ J.is_prime } :=
+begin
+  letI : Π i (x : 𝒜 i), decidable (x ≠ 0) := λ i x, classical.dec _,
+  rw ideal.radical_eq_Inf,
+  apply le_antisymm,
+  { refine Inf_le_Inf _,
+    rintros J ⟨HJ₁, _, HJ₂⟩,
+    exact ⟨HJ₁, HJ₂⟩, },
+  { intros x hx,
+    rw [submodule.mem_Inf] at hx ⊢,
+    rintros J ⟨HJ₁, HJ₂⟩,
+    refine (ideal.coe_homogeneous_core_le 𝒜 J) (hx _ _),
+    refine ⟨_, subtype.prop _, HJ₂.homogeneous_core⟩,
+    refine (homogeneous_ideal.homogeneous_core_coe_eq_self ⟨I, hI⟩).symm.trans_le
+      (ideal.homogeneous_core_mono _ HJ₁), }
+end
+
+lemma homogeneous_ideal.radical_eq (I : homogeneous_ideal 𝒜) :
+  (I : ideal A).radical = Inf {J | ↑I ≤ J ∧ J.is_homogeneous 𝒜 ∧ J.is_prime} :=
+I.prop.radical_eq
+
```

Comment:
Thinking again, I'd just get rid of this. `I.prop.radical_eq` is not just defeq to this statement, it's syntactically equal too: so you can just use `rw I.prop.radical_eq` anywhere you were previous using `rw I.radical_eq`.

Reply:
Thank you! Done.